### ISSUE TITLE
server.createList overrides argument is unintuitive

### ISSUE THREAD

**Issue by taras** [miragejs/miragejs#574]
I'm not sure if it's just me but I find createList overrides argument unintuitive.

From the docs, it looks like it's designed to apply the same values to all items it creates. 

```js
let contact = server.create('contact')

server.createList('address', 5, { contact })
```

From experimentation, I found that passing an array as the 3rd argument doesn't do anything. This makes it a little harder to handle a case when you have 6 items with different values that you want to prepopulate. For example,

```
let contact = server.create('contact')
let addresses = [
  {city: 'New York', contact}, 
  {city: 'San Francisco', contact}, 
  {city: 'Los Angeles', contact}, 
  {city: 'Detroit', contact}, 
  {city: 'Chicago', contact}
]; 

server.createList('address', 5, addresses)
```

Instead, you have to do something like this,

```
let addresses = [
  {city: 'New York'}, 
  {city: 'San Francisco'}, 
  {city: 'Los Angeles'}, 
  {city: 'Detroit'}, 
  {city: 'Chicago'}
]; 

addresses.forEach((address) => server.create('address', { contact, ...address});
```

It seems like it could be pretty straight forward to detect if override is an array and apply each item in the list as a unique override. Alternatively, API docs should show to use forEach or map to create multiple items.

*Note*: I know factories could do this better. In some cases, factores don't work well when you need an exact list in a specific order because someone else wrote tests like that :)

**Comment by samselikoff** [comment]
Yeah createList is really just a convenience for when the overrides are the same. It's mostly there because FactoryBot had a similar api `create_list`. I think if I were writing this today I wouldn't necessarily even include createList, just to keep the API smaller. Try to stick to ES idioms like map and `...`.

I'd probably write your example like

```js
let contact = server.create('contact')

server.create('address', { city: 'New York', contact })
server.create('address', { city: 'San Francisco', contact })
server.create('address', { city: 'Los Angeles', contact })
server.create('address', { city: 'Detroit', contact })
server.create('address', { city: 'Chicago', contact })
```

if I had an Address factory I wanted to make sure I was using, or

```js
let contact = server.create('contact')

contact.createAddress({ city: 'New York' })
contact.createAddress({ city: 'San Francisco' })
contact.createAddress({ city: 'Los Angeles' })
contact.createAddress({ city: 'Detroit' })
contact.createAddress({ city: 'Chicago' })
```

if I just wanted to create the raw relational data. Then I would abstract if duplication became a bigger issue.

I think `server.createList('address', 5, addresses)` is not quite justified as it's really just a bit of sugar, also what happens if the `addresses` array is not 5 items long? We could of course come up with behavior but, it's just one more thing for new users to learn & Mirage's API is already on the bigger side.

Just my initial thoughts!

**Comment by taras** [comment]
That makes sense. Maybe just add an example to createList on how to handle multiple overides because function signature has `any` for type but it's really not any. 

<img width="772" alt="image" src="https://user-images.githubusercontent.com/74687/89214935-b6169c80-d595-11ea-94dd-fe3848b588f5.png">

It might save others some time in the future.

**Comment by samselikoff** [comment]
Ah - yes absolutely! Need to go over the docs with a fine-toothed comb after we close bugs in prep for a v1.0.

Will mark this a bug and leave it open, thanks for pointing this out!
